javascript/ql/src/experimental/Security/CWE-326/ecb_rejection.ql

```diff
@@ -7,22 +7,21 @@
  */
 
 import javascript
-import semmle.javascript.Concepts
-import semmle.javascript.frameworks.CryptoLibraries
+/*import semmle.javascript.Concepts
+import semmle.javascript.frameworks.CryptoLibraries*/
 
 class WeakBlockMode extends BlockMode {
-    WeakBlockMode() {
-        exists(CryptographicOperation application |
-            (
-                application.getBlockMode().isWeak()
-            ) and
-            this = application.getBlockMode()
-        )
-    }
+  WeakBlockMode() {
+    exists(CryptographicOperation application |
+      application.getBlockMode().isWeak() and
+      this = application.getBlockMode()
+    )
+  }
 }
 
 from BlockMode bm, WeakBlockMode wbm, CryptographicOperation op
 where
   op.getBlockMode() = bm and
   bm = wbm
-select "A weak block cipher mode like " + bm + " does not secure sensitive data.", "The data is encoded with the weak block cipher in op. Make sure it is not sensitive data.", op
\ No newline at end of file
+select "A weak block cipher mode like " + bm + " does not secure sensitive data.",
+  "The data is encoded with the weak block cipher in $@. Make sure it is not sensitive data.", op , "this cryptographic operation"
```

Comment:
```suggestion
select "A weak block cipher mode like " + bm + " does not secure sensitive data. The data is encoded with the weak block cipher in $@. Make sure it is not sensitive data.", op , "this cryptographic operation"
```

Having multiple strings with a comma in between them like that doens't work well.  
The code that interprets the alert expects there to be one string, and not multiple.  

It's only if you use `$@` that multiple strings are valid, and there the extra strings are used as the link texts.

Reply:
Thank you. Unfortunately this is still not seeming to link the op properly. In the output, I get "$@" in the string and a separate column with op and a third column with "this cryptographic operation"

Reply:
Ahh, that's because you are missing the `@kind` in the header.  
I added a suggestion.